pyproject.toml

```diff
@@ -33,6 +33,7 @@ dependencies = [
     "rdkit",
     "scikit-learn",
     "scipy",
+    "tensorboard",
```

Comment:
tensorboard might already be an implicit dependency through pytorch lightning. My conda environment already had `tensorboardx=2.6.2` without my specification. This [file](https://github.com/Lightning-AI/pytorch-lightning/blob/93c1ab0653d50db1fb6c3cd3a7c26f4cee5cecac/requirements/pytorch/test.txt#L18) may be related.

Reply:
That's weird, because the github workflow fails due to the dependency issue [here](https://github.com/chemprop/chemprop/actions/runs/7574807468/job/20630020134#step:6:28) when testing the CLI part, and that's why I added it as a dependency explicitly.

Reply:
@JacksonBurns Any idea why we need to be explicit with the tensorboard dependency for the testing while when I installed from conda, it got automatically installed with lightning?

Reply:
we don't need it for the base package but do require it for the CLI. So your new CI is failing because you're trying to run the CI during testing.

Reply:
@KnathanM lightning used to ship with tensorboard (like, when you made your local environment), but they made it optional (so the CI that runs today won't include it). See this discussion if you're curious for more: https://github.com/Lightning-AI/pytorch-lightning/issues/9900

Reply:
That is exactly what I wanted to know. Thank @JacksonBurns! 
If lightning made tensorboard optional and defaults to CSVLogger when tensorboard isn't available, do we want to do the same? This would involve checking if the environment has tensorboard, which shouldn't be too difficult and would allow us to remove the tensorboard dependency.